library/Rules/AbstractRule.php

```diff
@@ -18,7 +19,7 @@ public function __construct()
 
     public function __invoke($input)
     {
-        return !is_a($this, __NAMESPACE__.'\\NotEmpty')
+        return !$this instanceof RequiredValidatable
             && $input === '' || $this->validate($input);
```

Comment:
The problem is, we have to inject this rule here, otherwise we will create one more rule to each rule we have in a chain.

Probably we can do this after the API change.